From ccba8f70fad5ce723e2db990c7fc577c680a2833 Mon Sep 17 00:00:00 2001
From: Jan Andersson <jan@gaisler.com>
Date: Tue, 8 Feb 2011 17:51:19 +0100
Subject: [PATCH 6/9] SPARC32: Add low level in_/out_ routines

This patch defines {in,out}_{8,le16,be16,le32,be32} on SPARC32. Some
drivers makes use of these low level I/O routines found on PowerPC in
linux/arch/powerpc/boot/io.h and also on other architectures.

Signed-off-by: Jan Andersson <jan@gaisler.com>
---
 arch/sparc/include/asm/io_32.h |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/arch/sparc/include/asm/io_32.h b/arch/sparc/include/asm/io_32.h
index c2ced21..e437646 100644
--- a/arch/sparc/include/asm/io_32.h
+++ b/arch/sparc/include/asm/io_32.h
@@ -267,6 +267,21 @@ _memcpy_toio(volatile void __iomem *dst, const void *src, __kernel_size_t n)
 
 #define memcpy_toio(d,s,sz)	_memcpy_toio(d,s,sz)
 
+/*
+ * Low-level I/O routines
+ * Modeled after PowerPC Low-level I/O routines
+ */
+#define in_8(__addr)		__readb(__addr)
+#define out_8(__addr, __b)	__writeb((__b), (__addr))
+#define in_le16(__addr)		__readw(__addr)
+#define in_be16(__addr)		__raw_readw(__addr)
+#define out_le16(__addr, __w)	__writew((__w), (__addr))
+#define out_be16(__addr, __w)	__raw_writew((__w), (__addr))
+#define in_le32(__addr)		__readl(__addr)
+#define in_be32(__addr)		__raw_readl(__addr)
+#define out_le32(__addr, __l)	__writel((__l), __addr)
+#define out_be32(__addr, __l)	__raw_writel((__l), (__addr))
+
 #ifdef __KERNEL__
 
 /*
-- 
1.5.4

